### PR TITLE
feat: adding jsdoc linting

### PR DESCRIPTION
**Pull request by Kreeg**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +104,14 @@
       ],
       "eslint-comments/disable-enable-pair": "off",
       "guard-for-in": "off",
+      "jsdoc/check-values": [
+        "warn",
+        {
+          "allowedLicenses": ["MIT https://github.com/svg-sprite/svg-sprite/blob/main/LICENSE"]
```
Comment (Kreeg) [inline package.json:110]: there is 2 options here
1. keep this
2. change license to "MIT" (see https://spdx.org/licenses/)

File: package.json [hunk 2]
```diff
@@ -97,6 +104,14 @@
...
+        }
+      ],
+      "jsdoc/no-undefined-types": ["error", {"definedTypes": ["SVGSpriter", "File", "SVGShape", "SVGSprite", "puppeteer", "HTMLElement"]}],
```
Comment (Kreeg) [inline package.json:113]: Or we can provide @typedef

File: lib/svg-sprite.js [hunk 3]
```diff
@@ -30,7 +29,8 @@ const CPU_COUNT = os.cpus() && os.cpus().length;
 /**
  * SVGSpriter class
  *
- * @param {Object} config                Configuration
+ * @param {object} config                Configuration
+ * @class
```
Comment (Kreeg) [inline lib/svg-sprite.js:33]: Not sure about this

**Comment by XhmikosR** [comment]
Not a huge fan of this myself but if it helps you, go ahead. That being said, a warning might be moot since the build won't fail. On the other hand, failing the build for this stuff might be too much.

PS. Unless you CC me I don't get notified. I have too many projects :/

File: lib/svg-sprite.js (hunk 3 above)
Comment (XhmikosR) [reply lib/svg-sprite.js:33]: It's not a Class AFAICT. I tried in the past to switch to Classes but didn't succeed so I kept the prototype methods.

**Comment by XhmikosR** [comment]
Nitpicking: indentation seems off in many places, not sure if there's any way to automate this too :)

**Comment by Kreeg** [comment]
@XhmikosR indentation for this thing is available with prettier-jsdoc-plugin, but for now I'll eyeball it

**Pull request merged**